### PR TITLE
feat(capture): ordered create-if-not-found location for reverse-chrono logs (#481)

### LINKED ISSUE #481: [FEATURE REQUEST] combining multiple insert-afters
**Is your feature request related to a problem? Please describe.**
I am trying to create a Capture document that has an H1 title, followed by a bit of blurb at the top; then an H2 dated heading. When making a capture, I want it placed BELOW that H2 dated heading, with subsequent captures from that day directly under the date heading (ie at the TOP of that 'section'). On subsequent days, I want that day's date-heading placed ABOVE the previous date. 
The captures should be placed in order from most recent to oldest; followed by another H2 dated heading, followed by captures flowing, etc, etc, 

**Describe the solution you'd like**
Christian suggested this on the Obs discord: "some sort of 'insert date heading after title heading if the date heading doesn't exist. If it does, or after creating it, insert the captured value'. Like combining multiple insert-afters."

**Additional context**
This is a mock-up of what I am after: 
image

### PR DESCRIPTION
**Pull request by chhoumann**
## Summary

Adds a fourth **"Create line if not found"** location to Capture's *Insert after* — **Ordered** — that creates a missing target heading at its **sorted position among same-level sibling headings**, instead of only at the top/bottom/cursor.

This is the generic primitive behind the long-standing request in #481 ("combining multiple insert-afters"): a reverse-chronological daily log where a fixed title/intro stays pinned, each new day's `## YYYY-MM-DD` heading is auto-created **below the intro and above older days**, and the newest entry sits at the top of each day. The same one mechanism also covers semver changelogs, year-grouped lists, alphabetical/numeric section logs, etc. via configuration.

Closes #481.

## How it works

When the insert-after heading (e.g. `## {{DATE:YYYY-MM-DD}}`) is missing, the new `Ordered` location sorts it into place among its same-level siblings:

- **`SectionOrdering`** (`by`: insertion / lexical / numeric / date / semver, `direction`: asc/desc, `dateFormat`, `unparseable`) — read **only** when `createIfNotFoundLocation === "ordered"`. Fully additive: no migration, no version bump; existing choices are byte-identical.
- A pure, **frontmatter- and fence-aware** placement helper (`orderedSectionPlacement.ts`).
- A **byte-preserving** splice (existing line endings untouched → minimal diff).
- **Idempotent**: the heading is created once; later captures find it and use the normal found-path. Within-day ordering uses the existing `Insert at end of section` toggle.
- Builder UI: a 4th create-location option + an *ordered* sub-panel; `Ordered` is mutually exclusive with `Inline insertion`.

Requirement structure falls out of the design: the title/intro stays pinned because it's a different heading level (and frontmatter is excluded entirely); the day heading is the ordered create; within-day order is the existing `insertAtEnd`.

## Example (#481)

```markdown
# My Daily Log

A short intro that always stays at the top.

## 2026-06-16
- 09:40 reviewed the code
- 09:13 started the design

## 2026-06-14
- 09:00 older entry
```

## Testing

- **51 new tests** (pure-helper placement matrix, full-handler integration, Load migration) — full suite green (**2368 passing**); `tsc`, `svelte-check`, and `eslint` clean; docs site builds.
- **End-to-end verified in a real Obsidian vault** (isolated worktree instance): the #481 daily log (incl. rendered screenshot), a semver changelog, the Keep-a-Changelog `[1.10.0]` bracket form, frontmatter-preserving captures, mixed/CRLF EOL preservation, and idempotent repeat captures.

## Review

Iterated through multiple review rounds (multi-lens + cross-model adversarial). Fixes that came out of review include: frontmatter-aware placement (prevented a YAML-corruption path), byte-preserving mixed-EOL/EOF splice, multi-line-anchor de-duplication (idempotency), fence-aware heading detection, builder reactivity that never clobbers a saved sort config, and the documented requirement that the capture format end in a newline.

## Deferred (documented limitations)

- Sorting is over **all** same-level headings in the note (not scoped per parent section) — correct for the single-group layouts above; `scopeToParent` is a future extension.
- Full SemVer pre-release precedence (build/pre-release metadata is dropped from the comparable core).
- "Re-sort an existing note's sections" is a separate future command (this only places the new section).

## Docs

New **"Ordered section placement"** subsection in `docs/docs/Choices/CaptureChoice.md` (Next/unreleased) with the daily-log recipe, sort options, changelog & books-by-year examples, and the limits.

## Summary by CodeRabbit

* **New Features**
  * Added an **ordered placement** mode for “insert after” captures to create missing headings at the correct sorted position using configurable sort strategies (insertion order, text, number, date, and semver).
  * Updated the UI and capture-choice loading to support ordered options and enforce ordered vs inline compatibility.
  * Ordered inserts now display success messages referencing the resolved target heading.
* **Documentation**
  * Expanded capture documentation with detailed ordered placement rules, constraints, and examples.
* **Tests**
  * Added comprehensive coverage for ordered section placement logic, edge cases, and document formatting behavior.

### CODE REVIEW

File: src/formatters/captureChoiceFormatter.ts [hunk 1]
```diff
@@ -726,11 +767,187 @@ export class CaptureChoiceFormatter extends CompleteFormatter {
 			}
 		}
 
+		if (
+			this.choice.insertAfter?.createIfNotFoundLocation ===
+			CREATE_IF_NOT_FOUND_ORDERED
+		) {
+			return this.createInsertAfterOrdered(formatted, insertAfterLine);
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/formatters/captureChoiceFormatter.ts:781]: Good catch — fixed in 8d18ee70. The ordered insert-after search now runs on fence-masked, frontmatter-excluded lines (`maskNonBodyHeadingsForSearch`), so a single-line target that only appears in a code fence or frontmatter no longer matches as "found"; the fence/frontmatter-aware ordered create path runs and builds the real sorted section. Added regression tests for both the fenced and frontmatter cases.

File: src/types/choices/CaptureChoice.ts [hunk 2]
```diff
@@ -147,6 +152,27 @@ export class CaptureChoice extends Choice implements ICaptureChoice {
...
+					direction: "desc",
+					unparseable: "bottom",
+				};
+			}
+			// Ordered always creates a heading on its own line; inline same-line
+			// insertion is mutually exclusive (the inline create path can't honor
+			// "ordered" and would abort). Enforce the UI invariant here too so a
+			// hand-edited/imported inline+ordered choice can't reach that abort.
+			loaded.insertAfter.inline = false;
+		}
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/types/choices/CaptureChoice.ts:178]: Done in 8d18ee70 — `orderBy` is now backfilled only when `createIfNotFound` is on, so an inert ordered config isn't mutated on load. The inline-safety normalization is unchanged.
